net/DevExtreme.AspNet.Data/RemoteGrouping/RemoteGroupExpressionCompiler.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace DevExtreme.AspNet.Data.RemoteGrouping {
+
+    class RemoteGroupExpressionCompiler<T> : ExpressionCompiler {
+        const int MAX_GROUPS = 8;
```

Comment:
It's the number of type arguments in the `RemoteGroupKey` class. Could be increased.